src/Pos/Update/Poll/Pure.hs

```diff
@@ -0,0 +1,70 @@
+-- | Pure Poll
+
+module Pos.Update.Poll.Pure
+       ( PurePoll (..)
+       ) where
+
+import           Control.Lens               (to, (%=), (.=))
+import qualified Data.HashMap.Strict        as HM
+import qualified Data.HashSet               as HS
+import           System.Wlog                (CanLog, HasLoggerName, NamedPureLogger)
+import           Universum                  hiding (toList)
+
+import           Pos.Update.Poll.Class      (MonadPoll (..), MonadPollRead (..))
+import qualified Pos.Update.Poll.PollState  as Poll
+import           Pos.Update.Poll.Types      (DecidedProposalState (dpsDifficulty),
+                                             ProposalState (..),
+                                             UndecidedProposalState (upsSlot))
+
+newtype PurePoll a = PurePoll
+    { getPurePoll :: NamedPureLogger (State Poll.PollState) a
+    } deriving (Functor, Applicative, Monad, CanLog, HasLoggerName)
+
+instance MonadPollRead PurePoll where
+    getBVState bv = PurePoll $ use $ Poll.psBlockVersions . to (HM.lookup bv)
+    getProposedBVs = PurePoll $ use $ Poll.psBlockVersions . to (HM.keys)
+    getEpochProposers = PurePoll $ use $ Poll.psEpochProposers
+    getConfirmedBVStates = PurePoll $ use $ Poll.psBlockVersions . to (HM.toList)
+    getAdoptedBVFull = PurePoll $ use $ Poll.psAdoptedBV
+    getLastConfirmedSV an = PurePoll $ use $ Poll.psConfirmedBVs . to (HM.lookup an)
+    getProposal ui = PurePoll $ use $ Poll.psActiveProposals . to (HM.lookup ui)
+    getProposalsByApp an = PurePoll $ do
```

Comment:
I don't like how `getProposalByApp` function is written. It's hard to reason about. First some action defined with `do` inside `let` inside `do`. And only then this action is applied to some applicative, but befor unwrapping from `Maybe` took place. Something like this looks more prettier to me:

```haskell
getProposalsByApp an = PurePoll $ do
    delActiveProposalsIndices <- use Poll.psDelActivePropsIdx
    activeProposals           <- use Poll.psActiveProposals
    pure $ propGetByApp delActiveProposalsIndices activeProposals ?: []
  where
    propGetByApp :: ...
    propGetByApp = do ...
```